mods/d2k/rules/misc.yaml

```diff
@@ -152,3 +152,97 @@ wormspawner:
 	RenderEditorOnly:
 	BodyOrientation:
 	WormSpawner:
+
+upgrade.conyard:
+	Tooltip:
+		Name: Construction Yard Upgrade
+		Description: Upgrade for the Construction Yard
+	Buildable:
+		BuildPaletteOrder: 50
+		Prerequisites: conyard
+		Queue: Upgrade
+		BuildLimit: 1
+	Valued:
+		Cost: 1000
+	RenderSprites:
+		Image: conyard.harkonnen
+		RaceImages:
+			atreides: conyard.atreides
+			ordos: conyard.ordos
+			corrino: conyard.corrino
+
+upgrade.barracks:
+	Tooltip:
+		Name: Barracks Upgrade
+		Description: Upgrade for the Barracks
```

Comment:
It would be better if these were more descriptive, perhaps something like "Enables advanced unit production".